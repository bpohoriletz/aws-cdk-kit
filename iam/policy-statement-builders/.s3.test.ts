import S3PolicyStatementBuilder from './s3';

describe('new S3PolicyStatementBuilder', () => {
  let instance: S3PolicyStatementBuilder;

  beforeEach(() => {
    instance = new S3PolicyStatementBuilder();
  });

  test('has correct defaults set', () => {
    expect(instance).toMatchSnapshot();
  });

  test('.setRead grants read, read version access', () => {
    instance.setRead();

    expect(instance).toMatchSnapshot();
  });

  test('.setWrite grants write access', () => {
    instance.setWrite();

    expect(instance).toMatchSnapshot();
  });

  test('.addResource adds resource', () => {
    instance.addResource('*');

    expect(instance).toMatchSnapshot();
  });
});
