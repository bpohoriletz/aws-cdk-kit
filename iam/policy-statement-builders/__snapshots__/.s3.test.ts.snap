// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new S3PolicyStatementBuilder .addResource adds resource 1`] = `
S3PolicyStatementBuilder {
  "policyStatementProps": PolicyStatementProduct {
    "actions": [],
    "resources": [
      "*",
    ],
  },
}
`;

exports[`new S3PolicyStatementBuilder .setRead grants read, read version access 1`] = `
S3PolicyStatementBuilder {
  "policyStatementProps": PolicyStatementProduct {
    "actions": [
      "s3:ListBucket",
      "s3:ListBucketVersions",
      "s3:GetObject",
      "s3:GetObjectVersion",
    ],
    "resources": [],
  },
}
`;

exports[`new S3PolicyStatementBuilder .setWrite grants write access 1`] = `
S3PolicyStatementBuilder {
  "policyStatementProps": PolicyStatementProduct {
    "actions": [
      "s3:PutObject",
    ],
    "resources": [],
  },
}
`;

exports[`new S3PolicyStatementBuilder has correct defaults set 1`] = `
S3PolicyStatementBuilder {
  "policyStatementProps": PolicyStatementProduct {
    "actions": [],
    "resources": [],
  },
}
`;
